### ISSUE TITLE
Test /2dcontext/pixel-manipulation/2d.imageData.put.nonfinite.html is wrong

### ISSUE THREAD

**Issue by nox** [web-platform-tests/wpt#13393]
This test checks that passing `Infinity` etc to `ctx.imageData` raises a `TypeError`, but this method takes `long` values, not `unrestricted double` ones.

**Comment by Juanmihd** [comment]
Hey nox,

That's right, the type expected in that function is of type long, and the test is testing precisely that when receiving not a long value (such as inifinity) it should raise a TypeError.

https://html.spec.whatwg.org/multipage/canvas.html#2dcontext This is where the spec talks about the type being long, as you said.

So it is actually correct that this is being tested to ensure it fails with not long values that cannot be converted to long (such as inifnity).

**Comment by jdm** [comment]
Agreed, step 6 of https://heycam.github.io/webidl/#abstract-opdef-converttoint says this should in fact throw a TypeError.

**Comment by nox** [comment]
AFAIU, WebIDL will happily convert `Infinity` to `0` for `long` arguments. See for example how `document.childNodes.item(Infinity)` returns the first child of the document.

**Comment by jdm** [comment]
You're right; I was looking at the wrong branch. This should be hitting step 8 instead.

**Comment by Juanmihd** [comment]
Yes, it will happily convert Infinity to 0 for `long` arguments, but not for `[EnforceRange] long` arguments. There is probably a missmatch between the [WPT test](http://w3c-test.org/2dcontext/pixel-manipulation/2d.imageData.put.nonfinite.html) and the [2dCanvas interface](https://html.spec.whatwg.org/multipage/canvas.html#2dcontext), that can be fixed either by changing the WPT test to accept infinities, or changing the specs to expect `[EnforceRange] long` arguments.

I've just created an [issue](https://github.com/whatwg/html/issues/4399) to try to address that by changing the spec to receive `[EnforceRange] long` arguments. I don't know which one makes more sense, but both seem as valid solutions.

**Comment by fserb** [comment]
Either we change the signatures to be `[EnforceRange]` or we remove those tests.
Either way, I guess this bug is gone. We can continue the conversation on the issue @Juanmihd created.

**Comment by jdm** [comment]
I don't think closing the issue before we actually resolve it makes sense.
